pandas/core/generic.py

```diff
@@ -8987,26 +8998,46 @@ def _align_series(
                     other.index, how=join, level=level, return_indexers=True
                 )
 
-            left = self._reindex_indexer(join_index, lidx, copy)
+            if is_series:
+                left = self._reindex_indexer(join_index, lidx, copy)
+            elif join_index is None:
+                left = self.copy() if copy else self
+            else:
+                data = {
+                    c: algorithms.take_nd(
+                        self.__getitem__(c)._values,
+                        lidx,
+                        allow_fill=True,
+                        fill_value=None,
+                    )
+                    for c in self.columns
+                }
+                left = self._constructor(
+                    dict_to_mgr(
```

Comment:
As it turns out, it is actually possible. Makes the code clearer because I don't need to determine the `manager` type. Thanks for the hint.

Probably I used `dict_to_mgr` because I got something else wrong on the way.

Code updated.